### PR TITLE
Moved SQL into yml. Fix a unit test failure.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 1]
```diff
@@ -94,13 +92,13 @@ public void itShouldCreateExam() throws ParseException {
...
                                         .withSection504(true)
                                         .withT3ProgramType("t3")
                                         .build()
                         ).build());
 
-        examStudentCount = jdbcTemplate.queryForObject(ExamStudentSQl, Integer.class);
+        examStudentCount = jdbcTemplate.queryForObject(examStudentSQl, Integer.class);
         assertThat(examStudentCount).isEqualTo(1);
 
     }
```
Comment (agorina) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: @malaffoon, I think this test is broken in my last check in. I have no idea how it worked on my machine. I suspect I had some data in the DB that made it pass, but after I cleaned the DB and re-run it, it was failing.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/repository/impl/AssessmentRepositoryImpl.java [hunk 2]
```diff
@@ -19,17 +21,36 @@
...
+    private final String findOneByNaturalIdSql;
+    private final String findClaimForAsmtSql;
+    private final String createClaimsForAsmt;
+    private final String createSql;
+
+
     @Autowired
-    private NamedParameterJdbcTemplate jdbcTemplate;
+    public AssessmentRepositoryImpl(final NamedParameterJdbcTemplate jdbcTemplate,
+                                    @NotNull @Value("${assessment.findOneByNaturalId.sql}") final String findOneByNaturalIdSql,
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/repository/impl/AssessmentRepositoryImpl.java]: Hmm. I think we should find a better pattern here. I realize we don't want to be loading and parsing the SQL over and over but doesn't the properties system of the Spring framework already cache things? If you just asked for the property when executing the SQL, wouldn't it be fast? If not, perhaps a SqlService that handles loading and caching. That way we don't end up with repo impl constructors with dozens of parameters.

File: exam-processor/src/main/resources/exam-processor.sql.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
+assessmentType:
+  toCode:
+      sql: >
```
Comment (malaffoon) [inline exam-processor/src/main/resources/exam-processor.sql.yml]: This should be `sql: >-` to eliminate trailing newline.

File: exam-processor/src/main/resources/exam-processor.sql.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
+assessmentType:
```
Comment (malaffoon) [inline exam-processor/src/main/resources/exam-processor.sql.yml]: I think we want additional scoping for these properties? Sure, we know how we're using them but, if you look in the properties space, they will be scattered within it. As an example, the spring data source properties aren't just bare keys like 'username', they are all prefixed with 'spring.datasource.' 

Let's discuss what we want to do for this.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 5]
```diff
@@ -34,16 +34,15 @@
...
-                    "  (-3, 1, null, null, 'SOCK_LS');",
             "INSERT INTO asmt (id, natural_id, grade_id,type_id, subject_id, academic_year, name, label, version) VALUES\n" +
                     "  (-1, 'SBAC)SBAC-ICA-FIXED-G5E-COMBINED-2017-Winter-2016-2017', 5, 1, 2, 2016, 'SBAC-ICA-FIXED-G5E-COMBINED-2017', 'Grade 5 ELA', '9831');\n",
-
+            "INSERT INTO claim (id, asmt_id, min_score, max_score, code) VALUES\n" +
+                    "  (-1, -1, null, null, 'SBAC-2-W'),\n" +
+                    "  (-2, -1, null, null, 'SBAC-4-CR'),\n" +
+                    "  (-3, -1, null, null, 'SOCK_LS');",
     })
     public void itShouldCreateExam() throws ParseException {
```
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: This is a nice happy path test. What would happen if the same code was executed without all those insert statements? Is that behavior that hasn't been coded yet so you didn't want a test to demonstrate it?

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 6]
```diff
@@ -55,15 +54,14 @@ public void itShouldCreateExam() throws ParseException {
                     .withScaleScore(claim)
                     .withScaleScoreStdErr(claim + 1)
                     .build());
-
         }
 
-        final String ExamStudentSQl = "SELECT count(*) from exam_student where " +
-                "grade_id = 1 and student_id = -11 and school_id = -11 and iep = 1 and lep = 0 and section504 = 1 and " +
+        final String examStudentSQl = "SELECT count(*) from exam_student where " +
```
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: examStudentSql

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java (hunk 1 above)
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: I experienced the same thing: i pulled down latest develop and did a clean build and it worked. I then did a schema clean/migrate and the test failed. I guess i'll wait for you to merge all this.

File: exam-processor/src/main/resources/exam-processor.sql.yml (hunk 4 above)
Comment (agorina) [inline exam-processor/src/main/resources/exam-processor.sql.yml]: Let's discuss.

File: exam-processor/src/main/resources/exam-processor.sql.yml (hunk 3 above)
Comment (agorina) [inline exam-processor/src/main/resources/exam-processor.sql.yml]: Will change with the next commit, after we talk. Just do not want to change it twice.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java (hunk 5 above)
Comment (agorina) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: You are correct, it just has not been coded. I have mile to cover with the testing.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java (hunk 1 above)
Comment (agorina) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: Interesting...

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java (hunk 6 above)
Comment (agorina) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: :) will fix. thank you!

**Pull request merged**